components/stack/stack.vue

```diff
@@ -0,0 +1,204 @@
+<template>
+  <component
+    :is="stackElement"
+    :class="[
+      'd-stack',
+      stackResponsive(),
+      defaultDir(),
+      stackGap,
+    ]"
+  >
+    <!-- @slot Slot for main content -->
+    <slot />
+  </component>
+</template>
+
+<script>
+import { DT_STACK_DIR, DT_STACK_GAP, DT_STACK_RESPONSIVE_BREAKPOINTS } from './stack_constants';
+
+export default {
+  name: 'DtStack',
+
+  props: {
+    /**
+     * Set this prop to the direction to stack the items.
+     * You can override the default direction with 'default' key.
+     * All the undefined breakpoints will have 'default' value
+     */
+    dir: {
+      type: [String, Object],
+      default: 'column',
+    },
+
+    /**
+     * Set this prop to render stack as a specific HTML element.
+     */
+    as: {
+      type: String,
+      default: 'div',
+    },
+
+    /**
+     * Set this prop to have the space between each stack item
+     */
+    gap: {
+      type: String,
+      default: '0',
+    },
+  },
+
+  data () {
+    return {
+      DT_STACK_DIR,
+      DT_STACK_GAP,
+      DT_STACK_RESPONSIVE_BREAKPOINTS,
+    };
+  },
+
+  computed: {
+    stackDir () {
+      return this.dir;
+    },
+
+    stackGap () {
+      return DT_STACK_GAP[this.gap] ? `d-stack--gap-${this.gap}` : null;
+    },
+
+    stackElement () {
+      return this.as ? this.as : 'div';
+    },
+  },
+
+  methods: {
+    defaultDir () {
+      if (DT_STACK_DIR[this.stackDir]) {
+        /** String dir resolution */
+        return DT_STACK_DIR[this.stackDir] !== 'column' ? `d-stack--${DT_STACK_DIR[this.stackDir]}` : null;
+      } else {
+        /** Object dir resolution */
+        const { default: defaultStyle } = this.stackDir;
+
+        return defaultStyle !== 'column' ? `d-stack--${DT_STACK_DIR[defaultStyle]}` : null;
+      }
+    },
+
+    stackResponsive () {
+      return [
+        ...DT_STACK_RESPONSIVE_BREAKPOINTS.map((breakpoint) => {
+          return this.stackDir[breakpoint] ? `d-stack--${breakpoint}--${this.stackDir[breakpoint]}` : null;
+        })];
+    },
+  },
+
+};
+</script>
+
+<style lang="less">
+.d-stack {
+  --stack-gap: 0;
+  --stack-direction: column;
+
+  display: flex;
+  flex-direction: var(--stack-direction);
+  gap: var(--stack-gap);
+
+  &--column-reverse {
+    --stack-direction: column-reverse;
+  }
+
+  &--row {
+    --stack-direction: row;
+  }
+
+  &--row-reverse {
+    --stack-direction: row-reverse;
+  }
+
+  &--xl{
+    @media screen and (max-width: 1264px) {
+      &--column {
+        --stack-direction: column;
+      }
+
+      &--column-reverse {
+        --stack-direction: column-reverse;
+      }
+
+      &--row {
+        --stack-direction: row;
+      }
+
+      &--row-reverse {
+        --stack-direction: row-reverse;
+      }
```

Comment:
Since these nested media query values are always all the same we can just put them in a mixin to avoid repetition

```less
.direction-options() {
    &--column {
      --stack-direction: column;
    }
    &--column-reverse {
      --stack-direction: column-reverse;
    }
    &--row {
      --stack-direction: row;
    }
    &--row-reverse {
      --stack-direction: row-reverse;
    }
}
```

and then

```less
&--xl {
    @media screen and (max-width: 1264px) {
       .direction-options();
   }
}

&--lg {
    @media screen and (max-width: 980px) {
       .direction-options();
   }
}

// and so on

Reply:
Excelent +1